docs/decisions/0001-purpose-of-this-repo.rst

```diff
@@ -1,29 +1,212 @@
+
+Note that the source of truth about subsidy state is stored in other services - license-manager
+is the source of truth for licenses, and ecommerce is the source of truth about coupon codes.
+The enterprise-access service is intended to be the source of truth about requests and
+approvals/denials of those requests for one of the above subsidy types (and perhaps
+other sub-domains of accessing enterprise properties in the future).  The subsidy source-of-truth
+services are still responsible for subsidy state and performing actions around
+assigning or revoking enterprise subsidies.
 
 Decision
---------
+========
+
+This service will be the source-of-truth about requests of enterprise subsidies and approvals
+or denials thereof.  
+
+Subsidy Request learner experience
+----------------------------------
+An enterprise learner must have access to the learner portal
+with a customer that has at least one current, active Subscription Plan or Enterprise Coupon.
+Somewhere in the user interface, it is made clear to the learner
+that they may request enrollment in a specific course from their learning administrator.
+Upon approval, the enterprise-access system will call one of the subsidy assignment endpoints.
+The enterprise-access system is *not* responsible for creating an enterprise enrollment.  Rather,
+upon successful subsidy assignment, the subsidy system notifies the learner of the assigned subsidy,
+with which the user may enroll in a course contained in their enterprise's catalog.
+
+The learner must be linked to the enterprise customer,
+that is, an ``EnterpriseCustomerUser`` record must exist for this (user, customer)
+association.
+
+We'll display a request button in the UI even if there are no unassigned
+subsidies available for this customer. In the future, displaying such unfulfillable requests
+to the customer admin could act as a "nudge" to the admin to procure more subsidies.
+
+Subsidy request data
+--------------------
+We'll store the following data related to the subsidy request:
+
+* A UUID to uniquely identify the request record.
+* The learner making the request, in the form of their LMS user id (though perhaps their email address is also needed).
+* The associated enterprise customer UUID.
+* The course run identifier for which the learner is requesting enrollment.
+* When the request was made.
+* Whether the request was approved or denied.
+* When the request was approved or denied.
+* The assigned subsidy UUID which fulfilled the request.
+* The identifier of the admin who approved or denied the request.
+* A history table
+
+A suggested data model design is presented in https://github.com/openedx/license-manager/pull/374/
+Notable aspects of this suggested design:
+
+* Requests for different types of subsidies live in different models, e.g.
+  ``LicenseRequest`` and ``CouponCodeRequest``.
+* Request approvals and denials live in different models, e.g.
+  ``LicenseRequestApproval``, ``LicenseRequestDenial``, etc.
```

Comment:
I'm currently in the process of writing tickets to create these models, and what I'm actually leaning toward now is just two models: `LicenseRequest` and `CouponCodeRequest` that hold both the request and approval/denial data.

Reply:
I'll come back and update the ADR once we groom the tickets.